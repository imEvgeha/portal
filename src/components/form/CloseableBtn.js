import React, {Component} from 'react';
import t from 'prop-types';
import {Button} from 'reactstrap';

export default class CloseableBtn extends Component {

    static propTypes = {
        title: t.string,
        id: t.string,
        value: t.string,
        disabled: t.bool,
        highlighted: t.bool,
        style: t.object,

        onClick: t.func,
        onClose: t.func,
    };

    constructor(props) {
        super(props);
        this.state = {};
        this.close = this.close.bind(this);
    }

    closeStyle = {
        fontWeight: '800',
        fontSize: '1.4rem',
        position: 'absolute',
        padding: '0 10px 0 2px',
        right: '0px',
        top: '0px'
    };

    close(event) {
        event.preventDefault();
        event.stopPropagation();
        this.props.onClose();
    }

    render() {
        return (
            <Button outline={!this.props.highlighted} color="secondary"
                    id={this.props.id}
                    onClick={this.props.onClick}
                    title={this.props.title + this.props.value}
                    disabled={this.props.disabled}
                    style={{width: '100%', textAlign: 'left', position: 'relative', ...this.props.style}}>
                <div style={{width: '100%', overflow: 'hidden', paddingRight: '20px', textOverflow: 'ellipsis'}}>
                    <small><strong>{this.props.title}</strong>{this.props.value}</small>
                </div>
                <a href={'#'} style={this.closeStyle} onClick={this.close} id={this.props.id + '-close-btn'}><span aria-hidden="true">&times;</span></a>
            </Button>
        );
    }
}
