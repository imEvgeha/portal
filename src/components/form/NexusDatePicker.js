import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import t from 'prop-types';
import {validateDate} from '../../util/Validation';
import {store} from '../../index';

const FIRST_DELIMETER_POSITION = 2;
const SECOUND_DELIMETER_POSITION = 5;

class NexusDatePicker extends Component {

    static propTypes = {
        date: t.any,
        id: t.string,
        disabled: t.bool,

        onChange: t.func,
        handleKeyPress: t.func,
        onInvalid: t.func,
    };

    prevRawInput = '';
    prevDate = null;

    constructor(props) {
        super(props);
        this.state = {
            dirty: false,
            open: false
        };

        this.clear = this.clear.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleChangeRaw = this.handleChangeRaw.bind(this);
        this.handleOnBlur = this.handleOnBlur.bind(this);
        this.handleClickOutside = this.handleClickOutside.bind(this);
        this.setValid = this.setValid.bind(this);
        this.refDatePicker = React.createRef();
    }

    toggleOpen = () => {
        if (this.state.open && this.refDatePicker.current.state.open) {
            setTimeout(() => {
                this.refDatePicker.current.setOpen(false);
            }, 10);
        }
        this.setState({open: !this.state.open});
    };

    componentDidMount() {
        this.refDatePicker.current.input.onclick = this.toggleOpen;
    }

    componentDidUpdate() {
        if (this.props.date === '') {
            this.prevDate = '';
            this.handleChange(null);
        }
        if (this.props.date === null && this.prevDate !== null) {
            this.prevDate = this.props.date;
            this.clear();
        }
        this.prevDate = this.props.date;
    }

    clear() {
        this.refDatePicker.current.clear();
        this.setState({dirty: false});
        this.prevRawInput = '';
        this.setValid(true);
    }

    handleChange(date) {
        this.props.onChange(date);
        this.setState({dirty: true});
        this.setValid(true);
    }

    handleOnBlur(locale = 'en') {
        this.setState({dirty: true, open: false});
        if (this.refDatePicker.current.input.value && !validateDate(this.refDatePicker.current.input.value, locale)) {
            this.setValid(false);
        }
    }

    // Hack for fix https://github.com/Hacker0x01/react-datepicker/issues/730
    handleClickOutside() {
        this.refDatePicker.current.cancelFocusInput();
        this.refDatePicker.current.setOpen(false);
    }

    setValid(valid) {
        this.setState({invalidDate: !valid});
        if (this.props.onInvalid) {
            this.props.onInvalid(!valid);
        }
    }

    // Used by ref
    focus() {
        this.refDatePicker.current.input.focus();
    }

    handleChangeRaw(date) {
        let justDirty = false;
        if (date) {
            if ( (this.prevRawInput.length === FIRST_DELIMETER_POSITION - 1 && date.length === FIRST_DELIMETER_POSITION) ||
                    (this.prevRawInput.length === SECOUND_DELIMETER_POSITION - 1 && date.length === SECOUND_DELIMETER_POSITION)) {
                this.refDatePicker.current.input.value = this.refDatePicker.current.input.value + '/';
            }

            if ( (this.prevRawInput.length === FIRST_DELIMETER_POSITION +1 && date.length === FIRST_DELIMETER_POSITION) ||
                    (this.prevRawInput.length === SECOUND_DELIMETER_POSITION + 1 && date.length === SECOUND_DELIMETER_POSITION || date.length > 10)) {
                this.refDatePicker.current.input.value = this.refDatePicker.current.input.value.slice(0, -1);
                date = this.refDatePicker.current.input.value;
            }

            if (!this.state.dirty && date.length === 10 ) {
                this.setState({dirty: true});
                justDirty = true;
            }

            if (validateDate(date)) {
                this.handleChange(moment(date));
            } else if (this.state.dirty || justDirty) {
                this.setValid(false);
            }

        } else {
            this.handleChange(null);
            this.setState({dirty: false});
        }
        this.prevRawInput = this.refDatePicker.current.input.value;
    }

    render() {
        const {locale} = store.getState().locale;

        return (
            <DatePicker
                ref={this.refDatePicker}
                className={this.state.invalidDate ? 'text-danger' : ''}
                id={this.props.id}
                placeholderText={'Enter date'}
                selected={this.props.date ? moment(this.props.date) : null}
                showYearDropdown
                showMonthDropdown
                autoComplete={'off'}
                allowSameDay={true}
                onChange={this.handleChange}
                onChangeRaw={(event) => this.handleChangeRaw(event.target.value)}
                onBlur={() => this.handleOnBlur(locale)}
                todayButton={'Today'}
                disabled={this.props.disabled}
                customInput={<input onKeyPress={this.props.handleKeyPress} />}
                onClickOutside={this.handleClickOutside}
                locale={locale}
            />
        );
    }
}

export default NexusDatePicker;
