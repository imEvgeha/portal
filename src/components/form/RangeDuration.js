import t from 'prop-types';
import React from 'react';
import {INVALID_DURATION} from '../../constants/messages';
import moment from 'moment';
import {AvField, AvForm} from 'availity-reactstrap-validation';

export default class RangeDuration extends React.Component {

    static defaultProps = {
        validateRange: true
    }

    static propTypes = {
        id: t.string,
        value: t.object,
        displayName: t.string,
        disabled: t.bool,
        onFromDurationChange: t.func,
        onToDurationChange: t.func,
        onInvalid: t.func,
        handleKeyPress: t.func,
        hideLabel: t.bool,
        validateRange: t.bool
    };

    constructor(props) {
        super(props);
        this.state = {
            invalidStartDuration: '',
            invalidEndDuration: '',
            prevFromDuration: null,
            prevToDuration: null,
            invalidRange: ''
        };
        this.handleChangeStartDuration = this.handleChangeStartDuration.bind(this);
        this.handleChangeEndDuration = this.handleChangeEndDuration.bind(this);
        this.handleInvalid = this.handleInvalid.bind(this);
        this.refFirstInput = React.createRef();
    }

    componentDidUpdate() {
        if(this.props.validateRange) {
            if (this.props.value.from && this.props.value.to && moment(this.props.value.from) > moment(this.props.value.to)) {
                if (!this.state.invalidRange && !this.state.invalidStartDuration && !this.state.invalidEndDuration) {
                    this.props.onInvalid(this.wrongDurationRange(true));
                }
            } else {
                if (this.state.invalidRange) {
                    this.props.onInvalid(this.wrongDurationRange(false));
                }
            }
        }
    }

    handleChangeStartDuration({target}) {
        const duration = target.value;
        this.props.onFromDurationChange(duration);
        if(this.props.validateRange) {
            const invalidRange = this.wrongDurationRange(duration && this.props.value.to && moment(this.props.value.to) < duration);
            this.setState({invalidStartDuration: ''});
            this.props.onInvalid(this.state.invalidEndDuration || invalidRange);
        }
    }

    handleChangeEndDuration({target}) {
        const duration = target.value;
        this.props.onToDurationChange(duration);
        if(this.props.validateRange) {
            const invalidRange = this.wrongDurationRange(duration && this.props.value.from && moment(this.props.value.from) > duration);
            this.setState({invalidStartDuration: ''});
            this.props.onInvalid(this.state.invalidStartDuration || invalidRange);
        }
    }

    wrongDurationRange(wrong) {
        if(!this.props.validateRange){
            return false;
        }
        const invalid = wrong ? this.props.displayName + ' from should be less than to' : '';
        this.setState({invalidRange: invalid});
        return !!invalid;
    }

    handleInvalid(name, value) {
        if (value) {
            this.setState({['invalid' + name +'Duration']: INVALID_DURATION, invalidRange: ''});
            this.props.onInvalid(true);
        } else {
            this.setState({['invalid' + name +'Duration']: ''});
        }
    }

    focus() {
    }

    render() {
        return (
            <div style={{ maxWidth:'300px', minWidth:'300px', flex:'1 1 300px', margin:'0 10px'}}>
                { !this.props.hideLabel && <label htmlFor="dashboard-rights-search-start-duration-text">{this.props.displayName}</label>}
                <div className="row justify-content-around">
                    <div style={{width: '45%', paddingLeft: '8px'}}>
                        <AvForm>
                            <AvField
                                name={this.props.id + '-from'}
                                id={this.props.id + '-from'}
                                ref={this.refFirstInput}
                                value={this.props.value.from !== undefined ? this.props.value.from : ''}
                                onChange={this.handleChangeStartDuration}
                                onInvalid={(value) => {this.handleInvalid('Start', value);}}
                                disabled={this.props.disabled}
                                onKeyPress={this.props.handleKeyPress}
                                type="text"
                            />
                        </AvForm>
                        {this.state.invalidStartDuration && (
                        <small
                            className="text-danger ml-2"
                            style={{position: 'absolute'}}
                        >{this.state.invalidStartDuration}
                        </small>
)}
                        {this.state.invalidRange && (
                        <small
                            className="text-danger ml-2"
                            style={{position: 'absolute'}}
                        >{this.state.invalidRange}
                        </small>
)}
                    </div>
                    <div>_</div>
                    <div style={{width: '45%', paddingRight: '8px'}}>
                        <AvForm>
                            <AvField
                                name={this.props.id + '-from'}
                                id={this.props.id + '-to'}
                                value={this.props.value.to !== undefined ? this.props.value.to : ''}
                                onChange={this.handleChangeEndDuration}
                                onInvalid={(value) => {this.handleInvalid('End', value);}}
                                disabled={this.props.disabled}
                                onKeyPress={this.props.handleKeyPress}
                                type="text"
                            />
                        </AvForm>
                        {this.state.invalidEndDuration && (
                        <small
                            className="text-danger ml-2"
                            style={{position: 'absolute'}}
                        >{this.state.invalidEndDuration}
                        </small>
)}
                    </div>
                </div>
            </div>
        );
    }
}
