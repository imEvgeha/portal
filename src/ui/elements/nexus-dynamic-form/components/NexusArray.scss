@import '../../../../styles/atlaskit/mixins';
@import '../../../../styles/atlaskit/colors';

.nexus-c-array {
    padding: 0.75rem 1.25rem;
    margin-bottom: 5px;
    .nexus-c-array__add {
        text-align: right;
    }
    .nexus-c-array__objects {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
        align-content: space-between;

        > div {
            flex: 0 49.5%;
            border: 1px solid map-get($light-neutral-colors, light-neutral-1);
            box-sizing: border-box;
        }
    }
    .nexus-c-array__object {
        padding: 25px;
        margin-top: 15px;
    }
}

.nexus-c-array__modal {
    .nexus-c-array__modal-buttons {
        text-align: right;
    }
}
