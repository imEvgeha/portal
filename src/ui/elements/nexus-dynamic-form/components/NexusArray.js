import React, {useContext, useState} from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {default as AKForm} from '@atlaskit/form/Form';
import EditorCloseIcon from '@atlaskit/icon/glyph/editor/close';
import {get} from 'lodash';
import {NexusModalContext} from '../../nexus-modal/NexusModal';
import {CANCEL, DELETE, REMOVE_TITLE} from '../../nexus-tag/constants';
import {buildSection, getFieldConfig, renderNexusField, getProperValues} from '../utils';
import {VIEWS, DELETE_POPUP} from '../constants';
import './NexusArray.scss';

const NexusArray = ({
    name,
    view,
    data,
    fields,
    getValues,
    setFieldValue,
    schema,
    setDisableSubmit,
    confirmationContent,
}) => {
    const {openModal, closeModal} = useContext(NexusModalContext);
    // allData includes initialData and rows added
    const [allData, setAllData] = useState(data);

    const renderAddButton = () => {
        return (
            <Button onClick={openEditModal} className="nexus-c-dynamic-form__add-button">
                {`+ Add ${name}`}
            </Button>
        );
    };

    const onRemove = index => {
        console.log(index);
    };

    const handleRemove = index => {
        if (confirmationContent) {
            const actions = [
                {
                    text: CANCEL,
                    onClick: closeModal,
                    appearance: 'default',
                },
                {
                    text: DELETE,
                    onClick: () => {
                        onRemove(index);
                        closeModal();
                    },
                    appearance: 'danger',
                },
            ];
            openModal(`${DELETE_POPUP} ${confirmationContent}`, {title: REMOVE_TITLE, width: 'medium', actions});
        } else {
            onRemove(index);
        }
    };

    const renderObject = (object, index) => {
        return (
            <div key={index} className="nexus-c-array__object">
                <div className="nexus-c-tag__remove-button" onClick={() => handleRemove(index)}>
                    <EditorCloseIcon size="medium" />
                </div>
                {buildObject(fields, allData[index] || {}, index)}
            </div>
        );
    };

    const buildObject = (fields = {}, initialData, index) => {
        return (
            <>
                {Object.keys(fields).map(key => {
                    return (
                        !getFieldConfig(fields[key], 'hidden', view) && (
                            <div
                                key={`nexus-c-array__field ${key}`}
                                className={`nexus-c-array__field ${fields[key].className ? fields[key].className : ''}`}
                            >
                                {renderNexusField(
                                    `${name}[${index}].${key}`,
                                    view,
                                    getValues,
                                    initialData,
                                    fields[key]
                                )}
                            </div>
                        )
                    );
                })}
            </>
        );
    };

    const buildButtons = (dirty, submitting, reset) => {
        return (
            <div className="nexus-c-array__modal-buttons">
                <Button
                    type="submit"
                    className="nexus-c-array__submit-button"
                    appearance="primary"
                    isDisabled={!dirty || submitting}
                >
                    Confirm
                </Button>
                <Button
                    className="nexus-c-array__cancel-button"
                    onClick={() => {
                        reset();
                        closeModal();
                    }}
                >
                    Cancel
                </Button>
            </div>
        );
    };

    const handleOnSubmit = values => {
        const formData = getValues();
        const arrayData = get(formData, name) ? get(formData, name) : [];
        // including the new row
        const editedArray = [...arrayData, values];

        const properValues = editedArray.map(v => getProperValues(schema, v));

        setAllData(editedArray);
        setFieldValue(name, properValues);
        setDisableSubmit(false);
        closeModal();
    };

    const modalContent = () => {
        return (
            <div className="nexus-c-array__modal">
                <AKForm onSubmit={values => handleOnSubmit(values)}>
                    {({formProps, dirty, submitting, reset, getValues}) => (
                        <form {...formProps}>
                            {buildSection(fields, getValues, VIEWS.CREATE, {})}
                            {buildButtons(dirty, submitting, reset)}
                        </form>
                    )}
                </AKForm>
            </div>
        );
    };

    const openEditModal = () => {
        openModal(modalContent(), {
            title: <div className="nexus-c-array__modal-title">{`Add ${name} Data`}</div>,
            width: 'medium',
        });
    };

    return (
        <div className="nexus-c-array">
            <div className="nexus-c-array__add">{view !== VIEWS.VIEW && renderAddButton()}</div>
            <div className="nexus-c-array__objects">{allData.map((o, index) => renderObject(o, index))}</div>
        </div>
    );
};

NexusArray.propTypes = {
    schema: PropTypes.array.isRequired,
    name: PropTypes.string.isRequired,
    view: PropTypes.string,
    data: PropTypes.array,
    fields: PropTypes.object,
    getValues: PropTypes.func,
    setFieldValue: PropTypes.func,
    setDisableSubmit: PropTypes.func,
    confirmationContent: PropTypes.string,
};

NexusArray.defaultProps = {
    view: VIEWS.VIEW,
    data: [],
    fields: {},
    getValues: undefined,
    setFieldValue: undefined,
    setDisableSubmit: undefined,
    confirmationContent: null,
};

export default NexusArray;
