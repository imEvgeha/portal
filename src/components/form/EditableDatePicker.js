import React, { Component } from 'react';
import { Button } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import moment from 'moment';
import t from 'prop-types';
import NexusDatePicker from './NexusDatePicker';
import {INVALID_DATE} from '../../constants/messages';

class EditableDatePicker extends Component {

    static propTypes = {
        validate: t.func,
        value: t.string,
        displayName: t.string,
        disabled: t.bool,
        onChange: t.func,
        priorityDisplay: t.any
    };

    constructor(props) {
        const vodDate = props.value;
        super(props);
        this.state = {
            date: vodDate ? moment(vodDate) : moment(),
            datePickerStatus: false,
            errorMessage: '',
            submitStatus: false
        };

        this.handleShowDatePicker = this.handleShowDatePicker.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.handleCancelDatePicker = this.handleCancelDatePicker.bind(this);
        this.handleInvalid = this.handleInvalid.bind(this);
        this.submit = this.submit.bind(this);
        this.cancel = this.cancel.bind(this);
    }

    handleShowDatePicker(e) {
        e.preventDefault();
        if (!this.props.disabled) {
            this.setState({
                datePickerStatus: true,
                date: moment(this.state.date).isValid() ? moment(this.state.date) : moment(),
            });
        }
    }
    handleCancelDatePicker(e) {
        e.preventDefault();
        this.cancel();
    }

    cancel() {
        this.setState({
            date: moment(this.props.value),
            datePickerStatus: false,
            errorMessage: ''
        });
    }

    handleChange(date) {
        this.setState({ date: date, submitStatus: false, errorMessage: '' });
    }

    handleInvalid(invalid) {
        if (invalid) {
            this.setState({ errorMessage: INVALID_DATE, submitStatus: true });
        }
    }

    submit(date) {
        const validationError = this.props.validate(date);
        if (validationError !== undefined) {
            this.setState({
                errorMessage: validationError
            });
        } else {
            this.props.onChange(date, this.cancel);
            this.setState({
                datePickerStatus: false
            });
        }
    }

    render() {
        const displayFunc = (value)=>{
            return (<span
                       onClick={this.handleShowDatePicker}
                       className={'displayDate' + (this.props.disabled ? ' disabled' : '')}>
                       {value}
                   </span>);
        };

        const unfocusedRender = ()=>{
            if(this.props.priorityDisplay) {
                return displayFunc(this.props.priorityDisplay);
            } else {
                if(this.props.value) {
                    return displayFunc(this.state.date ? moment(this.state.date).format('L') : '');
                } else {
                    return(
                        <span
                            style={{ color: '#808080', cursor: 'pointer' }}
                            onClick={this.handleShowDatePicker}>
                            {this.props.disabled ? '' : 'Enter ' + this.props.displayName}
                        </span>
                    );
                }
            }
        };

        return (
            <div>
                {
                    this.state.datePickerStatus ?
                        <div>
                            <div className="dPicker" style={{ marginBottom: '5px' }}>
                                <NexusDatePicker
                                    id="dashboard-avails-search-start-date-text"
                                    date={this.state.date}
                                    onChange={this.handleChange}
                                    onInvalid={this.handleInvalid}
                                />
                            </div>
                            <div style={{ float: 'left' }}>
                                <Button
                                    className="dPButton"
                                    disabled={this.state.submitStatus}
                                    onClick={() => this.submit(this.state.date ? this.state.date.toISOString() : null)}
                                    color="success"><FontAwesome name='check' />
                                </Button>
                                <Button
                                    className="dPButton"
                                    onClick={this.handleCancelDatePicker}
                                    color="danger"><FontAwesome name='times' />
                                </Button>
                            </div>
                            {
                                this.state.errorMessage &&
                                <p style={{ color: 'red', float: 'left', width: '100%' }}>
                                    {this.state.errorMessage}
                                </p>
                            }
                        </div>
                        :
                        unfocusedRender()
                }
            </div>
        );
    }
}

export default EditableDatePicker;
