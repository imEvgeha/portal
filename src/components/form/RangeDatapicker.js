import t from 'prop-types';
import React from 'react';
import NexusDatePicker from './NexusDatePicker';
import {INVALID_DATE} from '../../constants/messages';
import moment from 'moment';
import {momentToISO} from '../../util/Common';

export default class RangeDatapicker extends React.Component {
    static propTypes = {
        id: t.string,
        value: t.object,
        displayName: t.string,
        disabled: t.bool,
        onFromDateChange: t.func,
        onToDateChange: t.func,
        onInvalid: t.func,
        handleKeyPress: t.func,
        hideLabel: t.bool
    };

    constructor(props) {
        super(props);
        this.state = {
            invalidStartDate: '',
            invalidEndDate: '',
            prevFromDate: null,
            prevToDate: null,
            invalidRange: ''
        };
        this.handleChangeStartDate = this.handleChangeStartDate.bind(this);
        this.handleChangeEndDate = this.handleChangeEndDate.bind(this);
        this.handleInvalid = this.handleInvalid.bind(this);
        this.refFirstDatePicker = React.createRef();
    }

    componentDidUpdate() {
        if (this.props.value.from && this.props.value.to && moment(this.props.value.from) > moment(this.props.value.to)) {
            if (!this.state.invalidRange && !this.state.invalidStartDate && !this.state.invalidEndDate) {
                this.props.onInvalid(this.wrongDateRange(true));
            }
        } else {
            if (this.state.invalidRange) {
                this.props.onInvalid(this.wrongDateRange(false));
            }
        }
    }

    handleChangeStartDate(date) {
        this.props.onFromDateChange(date ? momentToISO(date) : null);
        const invalidRange = this.wrongDateRange(date && this.props.value.to && moment(this.props.value.to) < date);
        this.setState({invalidStartDate: ''});
        this.props.onInvalid(this.state.invalidEndDate || invalidRange);
    }

    handleChangeEndDate(date) {
        this.props.onToDateChange(date ? momentToISO(date) : null);
        const invalidRange = this.wrongDateRange(date && this.props.value.from && moment(this.props.value.from) > date);
        this.setState({invalidEndDate: ''});
        this.props.onInvalid(this.state.invalidStartDate || invalidRange);
    }

    wrongDateRange(wrong) {
        const invalid = wrong ? this.props.displayName + ' from should be before to date' : '';
        this.setState({invalidRange: invalid});
        return !!invalid;
    }

    handleInvalid(name, value) {
        if (value) {
            this.setState({['invalid' + name +'Date']: INVALID_DATE, invalidRange: ''});
            this.props.onInvalid(true);
        } else {
            this.setState({['invalid' + name +'Date']: ''});
        }
    }

    focus() {
        this.refFirstDatePicker.current.focus();
    }

    render() {
        return (
            <div style={{ maxWidth:'300px', minWidth:'300px', flex:'1 1 300px', margin:'0 10px'}}>
                { !this.props.hideLabel && <label htmlFor="dashboard-avails-search-start-date-text">{this.props.displayName}</label>}
                <div className={'row justify-content-around'}>
                    <div style={{width: '45%', paddingLeft: '8px'}}>
                        <NexusDatePicker
                            id={this.props.id + '-from'}
                            date={this.props.value.from !== undefined ? this.props.value.from : ''}
                            ref={this.refFirstDatePicker}
                            onChange={this.handleChangeStartDate}
                            onInvalid={(value) => {this.handleInvalid('Start', value);}}
                            disabled={this.props.disabled}
                            handleKeyPress={this.props.handleKeyPress}
                        />
                        {this.state.invalidStartDate && <small className="text-danger ml-2"
                                                               style={{position: 'absolute'}}>{this.state.invalidStartDate}</small>}
                        {this.state.invalidRange && <small className="text-danger ml-2"
                                                           style={{position: 'absolute'}}>{this.state.invalidRange}</small>}
                    </div>
                    <div>_</div>
                    <div style={{width: '45%', paddingRight: '8px'}}>
                        <NexusDatePicker
                            id={this.props.id + '-to'}
                            date={this.props.value.to !== undefined ? this.props.value.to : ''}
                            onChange={this.handleChangeEndDate}
                            onInvalid={(value) => {this.handleInvalid('End', value);}}
                            disabled={this.props.disabled}
                            handleKeyPress={this.props.handleKeyPress}
                        />
                        {this.state.invalidEndDate && <small className="text-danger ml-2"
                                                               style={{position: 'absolute'}}>{this.state.invalidEndDate}</small>}
                    </div>
                </div>
            </div>
        );
    }
}
