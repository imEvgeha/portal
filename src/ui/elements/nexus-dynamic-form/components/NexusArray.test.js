import React from 'react';
import {Field as AKField} from '@atlaskit/form';
import {shallow} from 'enzyme';
import NexusArray from './NexusArray';

describe('NexusArray', () => {
    let wrapper = null;
    const props = {
        key: 'rightId',
        name: 'rightId',
        label: 'Right ID',
        type: 'string',
        isDisabled: true,
        defaultValue: 'rght_zrp8g',
    };

    beforeEach(() => {
        wrapper = shallow(<NexusArray {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a AKField', () => {
        const textField = wrapper.find(AKField);
        expect(textField.length).toEqual(1);
    });
});
