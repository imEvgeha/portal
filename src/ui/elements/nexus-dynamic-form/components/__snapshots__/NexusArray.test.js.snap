// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusArray should match snapshot 1`] = `
<div
  className="nexus-c-array "
>
  <Field
    isDisabled={false}
    isRequired={false}
    validate={[Function]}
  >
    <Component />
  </Field>
  <div
    className="nexus-c-array__add"
  >
    <WithAnalyticsContext(WithAnalyticsEvents(Button))
      className="nexus-c-dynamic-form__add-button"
      onClick={[Function]}
    >
      + Add territory
    </WithAnalyticsContext(WithAnalyticsEvents(Button))>
  </div>
  <div
    className="nexus-c-array__objects"
  >
    <div
      className="nexus-c-array__object"
      key="0"
    >
      <div
        className="nexus-c-array__remove-button"
        onClick={[Function]}
      >
        <EditorCloseIcon
          size="medium"
        />
      </div>
      <div
        className="nexus-c-array__field "
        key="nexus-c-array__field selected"
      >
        <NexusField
          dateType=""
          dependencies={Array []}
          formData={Object {}}
          id="undefined[0].selected"
          isReadOnly={true}
          isRequired={false}
          key="undefined[0].selected"
          label="Selected"
          labels={Array []}
          name="undefined[0].selected"
          optionsConfig={Object {}}
          path="territory.selected"
          selectValues={Object {}}
          tooltip={null}
          type="boolean"
          validation={Array []}
          validationError={null}
          view="EDIT"
        />
      </div>
      <div
        className="nexus-c-array__field "
        key="nexus-c-array__field country"
      >
        <NexusField
          dateType=""
          dependencies={Array []}
          formData={Object {}}
          id="undefined[0].country"
          isReadOnly={false}
          isRequired={false}
          key="undefined[0].country"
          label="Country"
          labels={Array []}
          name="undefined[0].country"
          optionsConfig={Object {}}
          path="territory.country"
          selectValues={Object {}}
          tooltip={null}
          type="string"
          validation={Array []}
          validationError={null}
          view="EDIT"
        />
      </div>
    </div>
    <div
      className="nexus-c-array__object"
      key="1"
    >
      <div
        className="nexus-c-array__remove-button"
        onClick={[Function]}
      >
        <EditorCloseIcon
          size="medium"
        />
      </div>
      <div
        className="nexus-c-array__field "
        key="nexus-c-array__field selected"
      >
        <NexusField
          dateType=""
          dependencies={Array []}
          formData={Object {}}
          id="undefined[1].selected"
          isReadOnly={true}
          isRequired={false}
          key="undefined[1].selected"
          label="Selected"
          labels={Array []}
          name="undefined[1].selected"
          optionsConfig={Object {}}
          path="territory.selected"
          selectValues={Object {}}
          tooltip={null}
          type="boolean"
          validation={Array []}
          validationError={null}
          view="EDIT"
        />
      </div>
      <div
        className="nexus-c-array__field "
        key="nexus-c-array__field country"
      >
        <NexusField
          dateType=""
          dependencies={Array []}
          formData={Object {}}
          id="undefined[1].country"
          isReadOnly={false}
          isRequired={false}
          key="undefined[1].country"
          label="Country"
          labels={Array []}
          name="undefined[1].country"
          optionsConfig={Object {}}
          path="territory.country"
          selectValues={Object {}}
          tooltip={null}
          type="string"
          validation={Array []}
          validationError={null}
          view="EDIT"
        />
      </div>
    </div>
  </div>
</div>
`;
